### PR TITLE
Handle tx events on polygon

### PR DESCRIPTION
**Pull request by garethfuller**
Changes proposed in this pull request:
- Add a useEthers composable with simple tx listener function similar to our blocknative equivalent.
- If network is polygon handle tx confirmations with ethers tx listener.

This is a stop-gap solution until we role our own tx handler which is also needed to support the Gnosis integration.

### CODE REVIEW

**Review by evgenyboxer — APPROVED** [review]

File: src/components/forms/pool_actions/InvestForm.vue [hunk 1]
```diff
@@ -566,19 +600,11 @@ export default defineComponent({
...
-            data.loading = false;
-          },
-          onTxCancel: () => {
-            data.loading = false;
-          },
-          onTxFailed: () => {
-            data.loading = false;
-          }
-        });
+        if (isPolygon.value) {
```
Comment (TomAFrench) [inline src/components/forms/pool_actions/InvestForm.vue:603]:
```suggestion
        if (!supportsBlocknative.value) {
```
Can we make this more generic? We'll possibly be on other networks which don't have block stove support soon. I'd probably swap the two branches as well if we make this change.

**Review by TomAFrench — COMMENTED** [review]
Looks good. Only one change I would make but we can trust that up later if we just want to get this in asap.

File: src/components/forms/pool_actions/InvestForm.vue (hunk 1 above)
Comment (garethfuller) [reply src/components/forms/pool_actions/InvestForm.vue]: Good point! https://github.com/balancer-labs/frontend-v2/pull/478/commits/6911dbe65b6f2244ca3569e84c5b991d66e89664

File: src/composables/trade/useTokenApproval.ts [hunk 2]
```diff
@@ -90,14 +93,22 @@ export default function useTokenApproval(tokenInAddress, amount, tokens) {
...
-  function approvalTxListener(hash: string) {
+  function txHandler(tx: TransactionResponse): void {
+    if (supportsBlocknative.value) {
+      blocknativeTxHandler(tx.hash);
+    } else {
+      ethersTxHandler(tx);
+    }
+  }
+
+  function blocknativeTxHandler(hash: string): void {
```
Comment (TomAFrench) [inline src/composables/trade/useTokenApproval.ts:111]: This is inconsistent with what we have in the `InvestForm` where both handlers receive a `TransactionResponse` and BN pulls out the hash internally.

File: src/composables/useBlocknative.ts [hunk 3]
```diff
@@ -1,12 +1,23 @@
+import { computed } from 'vue';
```
Comment (TomAFrench) [inline src/composables/useBlocknative.ts:1]:
```suggestion
import { computed, injected } from 'vue';
```

Can merge imports here

**Pull request merged**